test/cmdlineTests.sh

```diff
@@ -334,10 +334,10 @@ function test_via_ir_equivalence()
 
     msg_on_error --no-stderr "$SOLC" --ir-optimized --debug-info location "${optimizer_flags[@]}" "$solidity_file" |
         sed '/^Optimized IR:$/d' |
-        split_on_empty_lines_into_numbered_files $output_file_prefix ".yul"
+        split_on_empty_lines_into_numbered_files "$output_file_prefix" ".yul"
 
     for yul_file in $(find . -name "${output_file_prefix}*.yul" | sort -V); do
-        msg_on_error --no-stderr "$SOLC" --strict-assembly --asm "${optimizer_flags[@]}" "$yul_file" |
+        msg_on_error --no-stderr "$SOLC" --strict-assembly --asm "$yul_file" |
```

Comment:
The PR is now based on #14149. This resolves the "Stack too deep" issue but now I have to deal with differences in the produced assembly.

Reply:
Finally got it to work. I originally got almost there by doing the assembling via Standard JSON which lets me disable Yul optimizer. This is annoying though because I get lots of spurious diffing failures due to different whitespace (even though we tell `diff` to ignore whitespace, which is weird).

I could get it to work on top of Standard JSON by beating it with a hammer long enough if we really wanted to but since the issue mentioned adding options to control EVM asm optimizations, I decided to check if that will make things easier. It worked out quite nicely and removed any need to tweak this test beyond adding `--no-optimize-yul`.

So now this PR should be passing all tests. Still a draft because I need to prepare a proper PR to make `--no-optimize-yul` work (for now I only quickly hacked it in). I also want to get the small refactors to the CLI test merged separate as #14261.